lib/fluent/plugin/out_opensearch_data_stream.rb

```diff
@@ -31,12 +31,16 @@ def configure(conf)
 
       @client = client
       unless @use_placeholder
+        delay = 1
         begin
           @data_stream_names = [@data_stream_name]
           create_index_template(@data_stream_name, @data_stream_template_name, @host)
           create_data_stream(@data_stream_name)
         rescue => e
-          raise Fluent::ConfigError, "Failed to create data stream: <#{@data_stream_name}> #{e.message}"
+          log.info "Failed to create data stream, will retry in #{delay} second(s): <#{@data_stream_name}> #{e.message}"
+          sleep(delay)
+          delay *= 2
+          retry
```

Comment:
Retrying forever is harmful in the current Fluentd core mechanism.
Instead, I've committed another retrying mechanism in https://github.com/fluent/fluent-plugin-opensearch/pull/8/commits/47b42756a69e53043b8d6d9bd92d8546ea4a9bd6